File: frontend/src/types/invoice.ts
export interface InvoiceItemCreate {
  service_id?: number | null;
  description: string;
  quantity: number;
  unit_price: number;
  tax_rate_percent: number;
}

export interface InvoiceItem extends InvoiceItemCreate {
  id: number;
  line_subtotal: number;
  line_tax_amount: number;
  line_total: number;
  sort_order: number;
}

export interface Invoice {
  id: number;
  company_id: number;
  client_id: number;
  currency_id: number;
  issue_date: string;
  due_date: string;
  notes?: string | null;
  invoice_number: string;
  status: string;
  subtotal_amount: number;
  tax_amount: number;
  total_amount: number;
  amount_paid: number;
  amount_due: number;
  pdf_path?: string | null;
  created_at: string;
  updated_at: string;
  items: InvoiceItem[];
}

export interface InvoiceCreate {
  company_id: number;
  client_id: number;
  currency_id: number;
  issue_date: string;
  due_date: string;
  notes?: string | null;
  items: InvoiceItemCreate[];
}

File: backend/app/schemas/invoice.py
from datetime import date, datetime
from typing import List

from pydantic import BaseModel


class InvoiceItemCreate(BaseModel):
    service_id: int | None = None
    description: str
    quantity: float
    unit_price: float
    tax_rate_percent: float


class InvoiceItemRead(InvoiceItemCreate):
    id: int
    line_subtotal: float
    line_tax_amount: float
    line_total: float
    sort_order: int

    class Config:
        from_attributes = True


class InvoiceBase(BaseModel):
    company_id: int
    client_id: int
    currency_id: int
    issue_date: date
    due_date: date
    notes: str | None = None


class InvoiceCreate(InvoiceBase):
    items: List[InvoiceItemCreate]


class InvoiceRead(InvoiceBase):
    id: int
    invoice_number: str
    status: str
    subtotal_amount: float
    tax_amount: float
    total_amount: float
    amount_paid: float
    amount_due: float
    pdf_path: str | None
    created_at: datetime
    updated_at: datetime
    items: List[InvoiceItemRead]

    class Config:
        from_attributes = True

